Add migration guide for 2.9.0 (#751)
digital-asset/docs.daml.com: docs/2.9.0/docs/canton/usermanual/upgrading.rst

Upgrade to Release 2.9
^^^^^^^^^^^^^^^^^^^^^^

Protocol versions
"""""""""""""""""
The recommended protocol versions is 5 (see :ref:`here <protocol_version>` for more information about protocol versions).

Version 2.9 does not offer support for protocol versions 3 and 4.

> If your sync domain is running one of these protocol versions, you need to perform the upgrade described :ref:`below <one_step_migration>`.

Reviewer: ```suggestion
If your synchronizer is running one of these protocol versions, you must upgrade as described :ref:`below <one_step_migration>`.
```

Author: On 2.9, it seems we are still using sync domain (and synchronizer will be used in 3.0)

Protocol version should be set explicitly
"""""""""""""""""""""""""""""""""""""""""

> Until now, the configuration of a domain was picking the latest protocol version by default.

Reviewer: ```suggestion
Until now, the synchronizers were configured to pick the latest protocol version by default.
```

Author: Same as above. Will use "sync domain"

> Since the protocol version is an important parameter of the domain, having this value set behind
> the scenes caused unwanted behavior.

Reviewer: ```suggestion
Since the protocol version is an important parameter, having this value set behind
the scenes caused unwanted behavior.
```

> You now must specify the protocol version for your domain:

Reviewer: ```suggestion
You now must specify the protocol version for your synchronizer:
```

Note that the ``migrate_domain`` command in 2.9 now automatically removes such data,
but only for the sync domain on which it has been invoked.

Paging in party Management
""""""""""""""""""""""""""
By default, the `ListKnownParties` method on the `PartyManagementService` will now only return at most 10'000
to avoid memory issues in participants that knows more than 10'000 parties.

The `next_page_token` can be used to request the next page (see `request <https://docs.daml.com/2.9.0/app-dev/grpc/proto-docs.html#com-daml-ledger-api-v1-admin-listknownpartiesrequest>`_
and `response <https://docs.daml.com/2.9.0/app-dev/grpc/proto-docs.html#com-daml-ledger-api-v1-admin-listknownpartiesresponse>`_).

Executor Service Metrics removed
""""""""""""""""""""""""""""""""

> The metrics for the execution services have been removed:

Reviewer 2: @mziolekda @atriantafyllos-da are there alternative metrics a customer should now monitor given these got removed? or the customer is supposed to remove those metrics from their monitoring? in the latter case, isn't there a blind spot?

Reviewer 3: The metrics were behind a quite prohibitive flag since 2.7 and got removed in 2.8. So this issue should not be mentioned on the 2.9 upgrading manual.

Author: OK, thanks, will remove.
